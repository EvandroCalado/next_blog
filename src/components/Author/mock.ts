export default {
  title: 'Evandro Calado',
  resume: 'Desenvolvedor Frontend',
  image: 'https://avatars.githubusercontent.com/u/110628201?v=4',
  alt: 'Evandro Calado',
};
