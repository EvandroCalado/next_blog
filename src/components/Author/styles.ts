import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
${({ theme }) => css`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 3rem;
  margin: ${theme.spacings.xxlarge} 0;
`}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 15rem;
    border: 3px solid ${theme.colors.secondary};
    border-radius: 50%;
  `}
`;

export const Container = styled.div`
`;

export const Title = styled.h2`
${({ theme }) => css`
font-size: ${theme.font.sizes.xlarge};
`}
`;

export const Paragraph = styled.p`
`;
