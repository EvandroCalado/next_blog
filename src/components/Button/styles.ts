import styled, { css } from 'styled-components';
import { ButtonProps } from './';

export const Button = styled.button<Pick<ButtonProps, 'color'>>`
  ${({ theme, color }) => css`
    background-color: ${theme.colors[color]};
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.medium};
    border: none;
    padding: ${theme.spacings.xsmall} ${theme.spacings.medium};
    margin: ${theme.spacings.large} 0;
    border-radius: 0.5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all 300ms ease-in-out;
    cursor: pointer;

    &:hover {
      filter: brightness(80%);
    }

    &:disabled {
      background-color: ${theme.colors.mediumGray};
      color: ${theme.colors.darkGray};
      cursor: not-allowed;

      &:hover {
        filter: none;
      }
    }

    > svg {
      width: 3rem;
      margin-left: 1rem;
    }
  `}
`;
