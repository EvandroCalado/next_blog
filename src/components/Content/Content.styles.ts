import styled, { css } from 'styled-components';

export const Wrapper = styled.article`
  ${({ theme }) => css`
    img {
      max-width: 100%;
    }

    p {
      margin: ${theme.spacings.medium} 0;
    }

    ul,ol {
      margin: ${theme.spacings.medium};
    }

    pre {
      width: 100%;
      overflow-x: auto;
      background-color: ${theme.colors.darkGray};
      color: ${theme.colors.white};
      border-radius: 0.5rem;
      padding: ${theme.spacings.large};
      margin: ${theme.spacings.large} 0;
      line-height: 1.5;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
