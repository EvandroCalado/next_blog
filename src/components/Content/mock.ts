export default {
  content:
    "<h1>Senior ignes duro</h1><h2>Ius ille Phlegyis</h2><p>Lorem markdownum Amore faciesque aethere volat perpetiar omine meriti antiquam!<strong>Illa</strong> est instruit hamos ulciscor sagitta rupta.</p><pre><code>pplWinTopology += intranet_drm.base_adsl_gate(standaloneExploitHdtv +        process_right_bloatware + archiveDelRom.dll_rte(88),        ripcording.gui_ppi(blog, cross_data_pum, barOf));wiki_prompt = gif_publishing(extranet_t_copy, virusNosql) + 5;if (dvdBox * 54) {    storageSoap += coreMpeg(xmp_storage, ssl) - 2 - hibernateSoundPrinter;} else {    teraflops.pmu += finder_megahertz_dvd + ttl_paste_website - 1 -            kerningWanBit;    tftp(pebibyte, webQuad + 3, 2);    radcab_market_php(faqScrollControl);}</code></pre><h2>Latens ad aut duc vetus qua</h2><p>Nec oscula fuerit: volet spectacula Quirini veluti, creditis cervice obambulatsaltem, est. Cycno tamen dixit est inrupit faciat circumdata: arvo laquei,<a href='http://equi.org/'>pudorem</a>! Et suos saltem fetus abstrahit elisi. In ignibusvocat: mediis tardae, manu <em>arida et ferebat</em>, si casuque diros ferrum, <a href='http://www.caput.com/de-levem.php'>nubilapro</a>. Sanctis dextra <em>somni equitavit ferox</em>et actaque iussa: duo in deo?</p><pre><code>rfid(imagePngLog(upPersonal, chip_pebibyte_microcomputer) - boot +        redundancy_ssd, 86);kvm = tokenResolution(7, friendly_component);if (1 == payload_name_state + pupPassiveOutput) {    abend_computing_recycle += hdmi +            webBps.grayscale_plug_bloatware.lanCpmDirect(search_key_pptp,            utfTeraflops, 830819);    platformBusRipcording(rateSecondaryMethod, runtime, domainRisc(            trinitron));    bit_newbie_video(torrent_certificate_file / 1);} else {    encoding += session;    megabit_drive_alpha = desktopServerClean;}</code></pre><h2>Est vidit sentit illa corpora ab matres</h2><p>Excussis pulsant navem, aeratas ea patria ac coniuge anhelitus verbis plusquemediaque in nondum. <strong>Et</strong> cogit Sinuessa tacetve, vicina calcataque haesurum,momenta ne, cedere.</p><h2>Portabat taurus saevorum tenuerunt imago Iuppiter texerat</h2><p>Inhibere electus visa notavi inde, conditur genualia viros et moderator tellusnumen? Negat pedibusque meum ignem in sub de dies prius exigit ad ultra illo?</p><blockquote><p>Nec velamine quid medias omne: honor omnes, os Iovis. Erat quater pedes nunc,arma agere, secuta ante? Elige vehit nulla abstulerat domus <em>adsumere</em> omniasaepe, et Dryope, et lumine?</p></blockquote><p>Vulnere ut erat violatus. Male illi terraeque significent viri, suoque, et etquae habere dubitavit temptamenta limina <a href='http://www.tendit-quamvis.net/ignisquecomminus'>vetustisparva</a> in domo talia. Nubesdignus neve transire <a href='http://www.esse-comparat.org/'>vocat adspice</a> femineo necmorique niveae solutum stagnata. <strong>Robore fert</strong>, glandes si viscera contigitintortos petis missus prius. Voluisti una supplex, iaculum satyri orbataqueetiam primum <em>venasque</em>, declivis Cydoneasque perstat, hominum fecit, nil.</p>",
};
