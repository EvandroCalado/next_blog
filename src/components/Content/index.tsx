import * as Styled from './styles';

export type ContentProps = {
  content: string;
};

const Content = ({ content }: ContentProps) => {
  return <Styled.Wrapper dangerouslySetInnerHTML={{ __html: content }} />;
};

export default Content;
