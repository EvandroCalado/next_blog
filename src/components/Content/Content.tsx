import * as Styled from './Content.styles';

export type ContentProps = {
  content: string;
};

const Content = ({ content }: ContentProps) => {
  return <Styled.Wrapper dangerouslySetInnerHTML={{ __html: content }} />;
};

export default Content;
