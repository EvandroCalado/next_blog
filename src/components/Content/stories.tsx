import Content, { ContentProps } from '.';
import { Meta, StoryFn } from '@storybook/react';

import mock from './mock';

export default {
  title: 'Content',
  component: Content,
  args: mock,
} as Meta;

export const Template: StoryFn<ContentProps> = (args) => (
  <div>
    <Content {...args} />
  </div>
);
