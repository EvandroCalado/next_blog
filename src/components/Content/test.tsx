import Content from './';
import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';

import mock from './mock';

describe('<Content />', () => {
  it('should render', () => {
    const { container } = renderTheme(<Content {...mock} />);

    expect(screen.getByRole('article')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
