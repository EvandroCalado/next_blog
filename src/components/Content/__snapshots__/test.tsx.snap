// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> should render 1`] = `
.c0 img {
  max-width: 100%;
}

.c0 p {
  margin: 2.4rem 0;
}

.c0 ul,
.c0 ol {
  margin: 2.4rem;
  color: #252422;
}

.c0 pre {
  width: 100%;
  overflow-x: auto;
  background-color: #333333;
  color: #f2f0d8;
  border-radius: 0.5rem;
  padding: 3.2rem;
  margin: 3.2rem 0;
  line-height: 1.5;
  font-size: 1.6rem;
}

<div>
  <article
    class="c0"
  >
    <h1>
      Senior ignes duro
    </h1>
    <h2>
      Ius ille Phlegyis
    </h2>
    <p>
      Lorem markdownum Amore faciesque aethere volat perpetiar omine meriti antiquam!
      <strong>
        Illa
      </strong>
       est instruit hamos ulciscor sagitta rupta.
    </p>
    <pre>
      <code>
        pplWinTopology += intranet_drm.base_adsl_gate(standaloneExploitHdtv +        process_right_bloatware + archiveDelRom.dll_rte(88),        ripcording.gui_ppi(blog, cross_data_pum, barOf));wiki_prompt = gif_publishing(extranet_t_copy, virusNosql) + 5;if (dvdBox * 54) {    storageSoap += coreMpeg(xmp_storage, ssl) - 2 - hibernateSoundPrinter;} else {    teraflops.pmu += finder_megahertz_dvd + ttl_paste_website - 1 -            kerningWanBit;    tftp(pebibyte, webQuad + 3, 2);    radcab_market_php(faqScrollControl);}
      </code>
    </pre>
    <h2>
      Latens ad aut duc vetus qua
    </h2>
    <p>
      Nec oscula fuerit: volet spectacula Quirini veluti, creditis cervice obambulatsaltem, est. Cycno tamen dixit est inrupit faciat circumdata: arvo laquei,
      <a
        href="http://equi.org/"
      >
        pudorem
      </a>
      ! Et suos saltem fetus abstrahit elisi. In ignibusvocat: mediis tardae, manu 
      <em>
        arida et ferebat
      </em>
      , si casuque diros ferrum, 
      <a
        href="http://www.caput.com/de-levem.php"
      >
        nubilapro
      </a>
      . Sanctis dextra 
      <em>
        somni equitavit ferox
      </em>
      et actaque iussa: duo in deo?
    </p>
    <pre>
      <code>
        rfid(imagePngLog(upPersonal, chip_pebibyte_microcomputer) - boot +        redundancy_ssd, 86);kvm = tokenResolution(7, friendly_component);if (1 == payload_name_state + pupPassiveOutput) {    abend_computing_recycle += hdmi +            webBps.grayscale_plug_bloatware.lanCpmDirect(search_key_pptp,            utfTeraflops, 830819);    platformBusRipcording(rateSecondaryMethod, runtime, domainRisc(            trinitron));    bit_newbie_video(torrent_certificate_file / 1);} else {    encoding += session;    megabit_drive_alpha = desktopServerClean;}
      </code>
    </pre>
    <h2>
      Est vidit sentit illa corpora ab matres
    </h2>
    <p>
      Excussis pulsant navem, aeratas ea patria ac coniuge anhelitus verbis plusquemediaque in nondum. 
      <strong>
        Et
      </strong>
       cogit Sinuessa tacetve, vicina calcataque haesurum,momenta ne, cedere.
    </p>
    <h2>
      Portabat taurus saevorum tenuerunt imago Iuppiter texerat
    </h2>
    <p>
      Inhibere electus visa notavi inde, conditur genualia viros et moderator tellusnumen? Negat pedibusque meum ignem in sub de dies prius exigit ad ultra illo?
    </p>
    <blockquote>
      <p>
        Nec velamine quid medias omne: honor omnes, os Iovis. Erat quater pedes nunc,arma agere, secuta ante? Elige vehit nulla abstulerat domus 
        <em>
          adsumere
        </em>
         omniasaepe, et Dryope, et lumine?
      </p>
    </blockquote>
    <p>
      Vulnere ut erat violatus. Male illi terraeque significent viri, suoque, et etquae habere dubitavit temptamenta limina 
      <a
        href="http://www.tendit-quamvis.net/ignisquecomminus"
      >
        vetustisparva
      </a>
       in domo talia. Nubesdignus neve transire 
      <a
        href="http://www.esse-comparat.org/"
      >
        vocat adspice
      </a>
       femineo necmorique niveae solutum stagnata. 
      <strong>
        Robore fert
      </strong>
      , glandes si viscera contigitintortos petis missus prius. Voluisti una supplex, iaculum satyri orbataqueetiam primum 
      <em>
        venasque
      </em>
      , declivis Cydoneasque perstat, hominum fecit, nil.
    </p>
  </article>
</div>
`;
